### PR TITLE
Moving LRO operation and writers into common so that it can be shared by mgmt.

### PR DESCRIPTION
**Pull request by allenjzhang**
# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [x] The PR has a understandable title and description explaining the _why_ and _what_.
- [x] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [x] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [x] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

**Review by pakrym — APPROVED** [review]

File: src/AutoRest.CSharp/Common/Output/Models/LongRunningOperation.cs [hunk 1]
```diff
@@ -14,17 +15,16 @@
...
-    internal class DataPlaneLongRunningOperation: TypeProvider
+    internal class LongRunningOperation: TypeProvider
     {
-        public DataPlaneLongRunningOperation(OperationGroup operationGroup, Input.Operation operation, BuildContext<DataPlaneOutputLibrary> context) : base(context)
+        public LongRunningOperation(OperationGroup operationGroup, Input.Operation operation, BuildContext context, LongRunningOperationInfo lroInfo) : base(context)
         {
             Debug.Assert(operation.IsLongRunning);
 
-            var clientClass = context.Library.FindClient(operationGroup);
+            //var lroInfo = context.Library.FindLongRunningOperationInfo(operationGroup, operation);
```
Comment (m-nash) [inline src/AutoRest.CSharp/Common/Output/Models/LongRunningOperation.cs:24]: nit: comment still needed?

**Review by m-nash — APPROVED** [review]
one minor comment

**Review by ShivangiReja — APPROVED** [review]

File: src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs [hunk 2]
```diff
@@ -196,6 +198,20 @@ public override CSharpType FindTypeForSchema(Schema schema)
...
         }
 
+        public LongRunningOperationInfo FindLongRunningOperationInfo(OperationGroup operationGroup, Operation operation)
+        {
+            var mgmtRestClient = FindRestClient(operationGroup);
+
+            var nextOperationMethod = operation.Language.Default.Paging != null
+                ? mgmtRestClient.GetNextOperationMethod(operation.Requests.Single())
+                : null;
+
```
Comment (markcowl) [inline src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs:208]: Feel like we need an assert over the client not being null here.  Also, null check on operation parameter before deref.

Comment (markcowl) [reply src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs:208]: And I think ```operation?.Language?.Default?.Paging``` is safest here.

**Review by markcowl — CHANGES_REQUESTED** [review]

**Review by markcowl — APPROVED** [review]

**Pull request merged**